src/processes/manager.rs

```diff
@@ -28,4 +37,28 @@ impl ProcessManager {
 
         processes.iter().find(|p| p.pid() == pid).cloned()
     }
+
+    fn execute(&self, executable: &Path, args: Vec<String>) -> Result<(), Error> {
```

Comment:
On second thought, we can  implement this as a `::new()` method for `::processes::Process`. Thoughts?

Reply:
Usually `new` is used for constructing a struct.

I would suggest renaming `execute` to `new` but move the logic to a new method with no arguments called `execute`.

Reply:
I mean, yeah that's the convention, but in our case there is no other point of creating a new instance of `Process` object.

Reply:
I'll need to review before I can comment further.